implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -166,6 +166,10 @@ private String decryptSignedToken(String token, JWTAuthContextInfo authContextIn
                             KeyEncryptionAlgorithm.RSA_OAEP.getAlgorithm()));
             jwe.setKey(getDecryptionKeyResolver(authContextInfo).resolveKey(jwe, null));
             jwe.setCompactSerialization(token);
+            if (!"JWT".equals(jwe.getContentTypeHeaderValue())) {
```

Comment:
@radcortez We don't need it, the mp text currently requires a JWT header somewhere but this headers does not bring any value in reality

Reply:
@radcortez Oh sorry this is the internal token

Reply:
No, we don't need still, on the builder API side we set this header but here it is too strict to enforce it

Reply:
This is to verify if the Content Type header is part of the token. If not, we fail the TCK test `org.eclipse.microprofile.jwt.tck.container.jaxrs.jwe.RolesAllowedSignEncryptTest#callEchoWithoutCty. 

Since we unwrap the Token before passing it to JOSE, it doesn't perform the content validation.

Reply:
@radcortez I forgot :-), OK, lets start with enforcing the best practice